gcore/gdal.h

```diff
@@ -1294,7 +1294,12 @@ GDALDriverH CPL_DLL CPL_STDCALL GDALGetDatasetDriver(GDALDatasetH);
 char CPL_DLL **CPL_STDCALL GDALGetFileList(GDALDatasetH);
 void CPL_DLL GDALDatasetMarkSuppressOnClose(GDALDatasetH);
 CPLErr CPL_DLL CPL_STDCALL GDALClose(GDALDatasetH);
+CPLErr CPL_DLL GDALCloseEx(GDALDatasetH hDS, GDALProgressFunc pfnProgress,
+                           void *pProgressData);
+bool CPL_DLL GDALDatasetGetCloseReportsProgress(GDALDatasetH hDS);
```

Comment:
About the API, if this is a driver's property (I see that for now only COG is reporting true) and it doesn't depend on a particular instance of the dataset, wouldn't be better to expose it as a driver capability flag?

Reply:
COG will actually not always report true. If you create the file with CreateCopy(), when CreateCopy() has completed, the file can be instantaneously closed. But for Create(), Close() will be lengthly. Hence a GDALDataset method. And this is an information you don't really need to know in advance.